x/concentrated-liquidity/bench_test.go

```diff
@@ -0,0 +1,235 @@
+	noError := func(err error) {
+		require.NoError(b, err)
+	}
+
+	const (
+		numberOfPositions = 10000
+
+		// max amount of each token deposited per position.
+		maxAmountDeposited = int64(1_000_000_000_000)
+
+		// amount swapped in.
+		amountIn = "9999999999999999999"
+
+		// flag controlling whether to create additional numberOfPositions full
+		// range positions for deeper liquidity.
+		shouldCreateFullRangePositions = true
+
+		// flag controlling whether to create positions concentrated around current tick to mimic
+		// realistic scenario and deeper liquidity.
+		// if true,
+		// creates numberOfPositions positions within 10 ticks of the current tick.
+		// creates numberOfPositions positions within 100 ticks of the current tick.
+		shouldConcentrate = true
+
+		// tickSpacing is the spacing between ticks.
+		tickSpacing = 1
+	)
+
+	var (
+		// denoms of the pool.
+		denom0 = DefaultCoin0.Denom
+		denom1 = DefaultCoin1.Denom
+
+		// denom of the token to swap in.
+		denomIn = denom0
+
+		numberOfPositionsInt = sdk.NewInt(numberOfPositions)
+		maxAmountOfEachToken = sdk.NewInt(maxAmountDeposited).Mul(numberOfPositionsInt)
+
+		// Seed controlling determinism of the randomized positions.
+		seed = int64(1)
+	)
+	rand.Seed(seed)
+
+	for i := 0; i < b.N; i++ {
+		s := BenchTestSuite{}
+		s.Setup()
+
+		// Full all accounts with max amounts they would need to consume.
+		for _, acc := range s.TestAccs {
+			simapp.FundAccount(s.App.BankKeeper, s.Ctx, acc, sdk.NewCoins(sdk.NewCoin(denom0, maxAmountOfEachToken), sdk.NewCoin(denom1, maxAmountOfEachToken), sdk.NewCoin("uosmo", maxAmountOfEachToken)))
+		}
+
+		// Craete a pool
+		poolId, err := s.App.PoolManagerKeeper.CreatePool(s.Ctx, clmodel.NewMsgCreateConcentratedPool(s.TestAccs[0], denom0, denom1, tickSpacing, sdk.MustNewDecFromStr("0.001")))
+		noError(err)
+
+		clKeeper := s.App.ConcentratedLiquidityKeeper
+
+		// Create first position to set a price of 1 and tick of zero.
+		tokenDesired0 := sdk.NewCoin(denom0, sdk.NewInt(100))
+		tokenDesired1 := sdk.NewCoin(denom1, sdk.NewInt(100))
+		tokensDesired := sdk.NewCoins(tokenDesired0, tokenDesired1)
+		_, _, _, _, _, err = clKeeper.CreatePosition(s.Ctx, poolId, s.TestAccs[0], tokensDesired, sdk.ZeroInt(), sdk.ZeroInt(), types.MinTick, types.MaxTick)
+
+		pool, err := clKeeper.GetPoolById(s.Ctx, poolId)
+		noError(err)
+
+		// Zero by default, can configure by setting a specific position.
+		currentTick := pool.GetCurrentTick()
+
+		// Setup numberOfPositions positions at random ranges
+		for i := 0; i < numberOfPositions; i++ {
+
+			var (
+				lowerTick int64
+				upperTick int64
+			)
+
+			if denomIn == denom0 {
+				// Decreasing price so want to be below current tick
+
+				// minTick <= lowerTick <= currentTick
+				lowerTick = rand.Int63n(currentTick.Int64()-types.MinTick+1) + types.MinTick
+				// Normalize lowerTick to be a multiple of tickSpacing
+				lowerTick = lowerTick + (tickSpacing - lowerTick%tickSpacing)
```

Comment:
```suggestion
```

Reply:
This is  incorrect

Reply:
I'm wrong. This is correct. I thought we must normalize prior to initializing upper tick but it doesn't matter. Normalization can happen fter